instrumentation-genai/opentelemetry-instrumentation-vertexai-v2/src/opentelemetry/instrumentation/vertexai_v2/__init__.py

```diff
@@ -0,0 +1,450 @@
+                )
+            )
+
+        # Copied from openllmetry logic
+        # https://github.com/traceloop/openllmetry/blob/v0.33.12/packages/opentelemetry-instrumentation-vertexai/opentelemetry/instrumentation/vertexai/__init__.py#L141-L143
+        # I guess prompt may be in kwargs instead or in addition?
+        prompt = kwargs.get("prompt")
+        if prompt:
+            event_logger.emit(
+                user_event(
+                    gen_ai_system=gen_ai_attributes.GenAiSystemValues.VERTEX_AI.value,
+                    content=prompt,
+                    span_context=span.get_span_context(),
+                )
+            )
+
+    _set_span_attribute(
+        span, gen_ai_attributes.GEN_AI_REQUEST_MODEL, llm_model
+    )
+    _set_span_attribute(
+        span,
+        gen_ai_attributes.GEN_AI_REQUEST_TEMPERATURE,
+        kwargs.get("temperature"),
+    )
+    _set_span_attribute(
+        span,
+        gen_ai_attributes.GEN_AI_REQUEST_MAX_TOKENS,
+        kwargs.get("max_output_tokens"),
+    )
+    _set_span_attribute(
+        span, gen_ai_attributes.GEN_AI_REQUEST_TOP_P, kwargs.get("top_p")
+    )
+    _set_span_attribute(
+        span, gen_ai_attributes.GEN_AI_REQUEST_TOP_K, kwargs.get("top_k")
+    )
+    _set_span_attribute(
+        span,
+        gen_ai_attributes.GEN_AI_REQUEST_PRESENCE_PENALTY,
+        kwargs.get("presence_penalty"),
+    )
+    _set_span_attribute(
+        span,
+        gen_ai_attributes.GEN_AI_REQUEST_FREQUENCY_PENALTY,
+        kwargs.get("frequency_penalty"),
+    )
+
+
+@dont_throw
+def _set_response_attributes(
+    span, event_logger: EventLogger, llm_model, generation_text, token_usage
+):
+    _set_span_attribute(
+        span, gen_ai_attributes.GEN_AI_RESPONSE_MODEL, llm_model
+    )
+
+    if token_usage:
+        _set_span_attribute(
+            span,
+            gen_ai_attributes.GEN_AI_USAGE_OUTPUT_TOKENS,
+            token_usage.candidates_token_count,
+        )
+        _set_span_attribute(
+            span,
+            gen_ai_attributes.GEN_AI_USAGE_INPUT_TOKENS,
+            token_usage.prompt_token_count,
+        )
+
+    if generation_text:
+        event_logger.emit(
+            assistant_event(
+                gen_ai_system=gen_ai_attributes.GenAiSystemValues.VERTEX_AI.value,
+                content=generation_text,
+                span_context=span.get_span_context(),
+            )
+        )
+
+
+def _build_from_streaming_response(
+    span, event_logger: EventLogger, response, llm_model
+):
+    complete_response = ""
+    token_usage = None
+    for item in response:
+        item_to_yield = item
+        complete_response += str(item.text)
+        if item.usage_metadata:
+            token_usage = item.usage_metadata
+
+        yield item_to_yield
+
+    _set_response_attributes(
+        span, event_logger, llm_model, complete_response, token_usage
+    )
+
+    span.set_status(Status(StatusCode.OK))
```

Comment:
we don't set span status in instrumentations to `OK` - only to `ERROR` if there is an error. In general case we don't know what constitutes a success. 

https://github.com/open-telemetry/opentelemetry-specification/blob/b9b6d8f698f82cd321fbe4fcc15b76d49902c14e/specification/trace/api.md?plain=1#L611